### PR TITLE
Add pages for picking up and viewing gem message

### PR DESCRIPTION
**Pull request by ming-00**
Still a lot of cleaning up to do and issues to resolve:
 - unfortunately protoc-gen-grpc-web doesn't support arm64 so I gotta find a workaround D: 
 - also trying to get vant popup component to work, but I think the bottom drawer will need to be implemented from scratch

### CODE REVIEW

**Comment by ianyong** [comment]
> unfortunately protoc-gen-grpc-web doesn't support arm64 so I gotta find a workaround D:

Does running terminal under Rosetta work?

**Comment by ming-00** [comment]
> Does running terminal under Rosetta work?

I tried that too but no go :P

**Comment by ming-00** [comment]
Thanks for being so patient with me y'all :-) Finally got to merge master branch in without problems... Please let me know if there are any dubious parts!

Changes:
 - Add pages for found gem
 - Added fonts and 1 video in

Next steps:
 - CSS clean up 👀 I can do the research for CSS methodologies!
 - Hoping to set up and standardise fonts 
 - Finish up UI for find gem flow (btw, for the pop up drawer, I'm unable to find a good method to take into account the relative height of the message div, rn I'm using max-content but it destroys the transition so it looks very abrupt... if there are any suggestions please lmk!)
 - Button appears to pick up gem on tooltip if within range
 - Update gem fields in protobuf + relevant backend tings

File: frontend/src/pages/PickupGem/FoundGem.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  components: { 
+    "van-button": Button,
+  },
+  computed: {
+    getPoster() {
+      // is this bad code, should i use switch instead :-)
+      if (Object.values(color).includes(this.gemPackage.color as color)) {
+        return require('@/assets/images/gem-' + this.gemPackage.color +'-found.png');
+      }
+      return require('@/assets/images/gem-purple-found.png');
```
Comment (sebastiantoh) [inline frontend/src/pages/PickupGem/FoundGem.vue]: looks good! For line 35, we don't expect this case to occur right? Maybe we can add an assert that `Object.values(color).includes(this.gemPackage.color as color)` then just `return require('@/assets/images/gem-' + this.gemPackage.color +'-found.png');`?

Likewise for the similar block of code below in the same file

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (sebastiantoh) [inline package-lock.json]: is this file necessary?

File: frontend/src/pages/PickupGem/FoundGem.vue [hunk 3]
```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="bg-container">
+    <video autoplay muted :poster="getPoster" id="gem-video">
+      <source :src="getVideo" type="video/mp4">
+    </video>
+    <div id="found-gem">
+      <h1>You found a gem! {{ gemPackage.color }}</h1>
```
Comment (sebastiantoh) [inline frontend/src/pages/PickupGem/FoundGem.vue]: Is it intentional that the color is mentioned here? Feels quite out of place haha

File: frontend/src/pages/PickupGem/MessageDisplay.vue [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<template>
+  <div class="bg-container" :style="addBackground">
+    <BottomDrawer :gemPackage="gemPackage"/>
+  </div>
+</template>
+<script  lang="ts">
+import { defineComponent, PropType } from "vue";
+import { GemInfo } from "@/interfaces";
+import BottomDrawer from "@/components/MessageBottomSheet.vue";
```
Comment (sebastiantoh) [inline frontend/src/pages/PickupGem/MessageDisplay.vue]: Is it intentional that BottomDrawer and MessageBottomSheet have different names?

**Review by sebastiantoh — DISMISSED** [review]
lgtm! just some minor comments. 

Also, for the popup drawer, I'm just wondering if it's necessary for us to collapse it to begin with?

File: frontend/src/interfaces/gem.ts [hunk 5]
```diff
@@ -9,3 +12,21 @@ export interface GemColor {
...
+export interface GemInfo {
+  message: string;
+  position: hansel.LatLng;
+  dropper: string;
+  receiver: string;
+  dropTime: Dayjs;
+  color: string;
+}
+
+export enum color {
```
Comment (okg00dbye) [inline frontend/src/interfaces/gem.ts:25]: nit: GemColor ?

File: frontend/src/interfaces/gem.ts [hunk 6]
```diff
@@ -9,3 +12,21 @@ export interface GemColor {
...
+  PURPLE = 'purple',
+  PINK = 'pink',
+  BLUE = 'blue',
+  BLACK = 'black',
```
Comment (okg00dbye) [inline frontend/src/interfaces/gem.ts:29]: feel free ignore: will black gem look unappetizing hahaha

File: frontend/src/pages/PickupGem/PickupGem.vue [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+enum PickupGemStage {
+  // eslint-disable-next-line no-unused-vars
+  FoundGem,
+  // eslint-disable-next-line no-unused-vars
+  MessageDisplay,
+}
+
+export default defineComponent({
+  setup() {
+    const gemPackage = {
```
Comment (okg00dbye) [inline frontend/src/pages/PickupGem/PickupGem.vue:20]: nit/feel free ignore: maybe can add a TODO here to replace it with fetched data or props

File: frontend/src/pages/PickupGem/FoundGem.vue (hunk 3 above)
Comment (ming-00) [reply frontend/src/pages/PickupGem/FoundGem.vue]: omg thanks for pointing it out LOL I think I was testing and then forgot to delete

**Review by okg00dbye — COMMENTED** [review]
I'm rlly sorry I merged seb's PR without realizing the consequences, you might have to update from master again D:

File: frontend/src/pages/PickupGem/FoundGem.vue (hunk 1 above)
Comment (ming-00) [reply frontend/src/pages/PickupGem/FoundGem.vue]: Yea, I wrote it as a fallback... okay, sounds good!

File: package-lock.json (hunk 2 above)
Comment (ming-00) [reply package-lock.json]: ay, lemme remove!

File: frontend/src/pages/PickupGem/MessageDisplay.vue (hunk 4 above)
Comment (ming-00) [reply frontend/src/pages/PickupGem/MessageDisplay.vue]: Nobody could decide what the component is called online... then I forgot to come to a conclusion D: Lemme standardise!

File: frontend/src/interfaces/gem.ts (hunk 6 above)
Comment (ming-00) [reply frontend/src/interfaces/gem.ts:29]: > I'm rlly sorry I merged seb's PR without realizing the consequences, you might have to update from master again D:

No worries!!

> feel free ignore: will black gem look unappetizing hahaha
 
I was gonna chope black gem for myself x(

File: frontend/src/pages/PickupGem/PickupGem.vue (hunk 7 above)
Comment (ming-00) [reply frontend/src/pages/PickupGem/PickupGem.vue:20]: Okay!

File: frontend/src/interfaces/gem.ts (hunk 6 above)
Comment (ming-00) [reply frontend/src/interfaces/gem.ts:29]: Screenshot 2021-09-19 at 12 06 50 AM
would you eat

File: frontend/src/interfaces/gem.ts (hunk 5 above)
Comment (okg00dbye) [reply frontend/src/interfaces/gem.ts:25]: oh I realized the clash, I'll update it in carousel PR

File: frontend/src/pages/PickupGem/FoundGem.vue (hunk 1 above)
Comment (ming-00) [reply frontend/src/pages/PickupGem/FoundGem.vue]: > Also, for the popup drawer, I'm just wondering if it's necessary for us to collapse it to begin with?

I think it's gonna be pretty big when fully expanded?

File: frontend/src/interfaces/gem.ts (hunk 6 above)
Comment (okg00dbye) [reply frontend/src/interfaces/gem.ts:29]: omg so spunky I love it. i think it will eat me

**Review by okg00dbye — APPROVED** [review]
lgtm, amazing animation!
